### PR TITLE
Update for testing rubin-env 8.

### PR DESCRIPTION
**Pull request by ktlim**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -159,21 +166,21 @@ outputs:
...
+        - pydantic >=2,<3
         - pyld
         - pysynphot
         - pytables
         - python
         - pytorch
         - pyyaml
         - requests
-        - ruff ==0.0.275
+        - ruff ==0.0.278
```
Comment (timj) [inline recipe/meta.yaml:177]: 0.1.7?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -144,13 +146,18 @@ outputs:
         - iminuit >=2
         - lmfit
         - lsstdesc.coord
+        - lsst-ts-xml
         - matplotlib-base !=3.6.0
+        - mkl  # [linux64]
```
Comment (erykoff) [inline recipe/meta.yaml:151]: Are we fixing to mkl on linux?  (May be fine, but not necessary I don't think).

**Review by erykoff — COMMENTED** [review]
We also want to make sure we migrate cfitsio, log4cxx, and boost, and check compiler versions if we can upgrade.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -144,13 +146,18 @@ outputs:
...
         - moto >=3
         - mpi4py
+        - nest_asyncio
         - networkx
+        - ngmix-core
         - numexpr =2
         - {{ pin_compatible('numpy') }}
-        - pandas
+        - panda-client !=1.5.65
+-       - pandas
```
Comment (erykoff) [inline recipe/meta.yaml:160]: Is this the lint?  Hard to see in the diff...

File: recipe/meta.yaml [hunk 4]
```diff
@@ -217,27 +224,30 @@ outputs:
...
         - bqplot
         - ciso8601
         - cloudpickle
         - cookiecutter
-        - dash
-        - dask
-        - dask-kubernetes
-        - dask_labextension
+        - dash >=2.11
+-       - dask-core
```
Comment (erykoff) [inline recipe/meta.yaml:235]: More lint

File: recipe/meta.yaml (hunk 2 above)
Comment (ktlim) [reply recipe/meta.yaml:151]: We have it in 7.0.1, so was just carrying it over.  Remove it?

File: recipe/meta.yaml (hunk 3 above)
Comment (ktlim) [reply recipe/meta.yaml]: Yes, I missed it.  Fixed now.

File: recipe/meta.yaml (hunk 1 above)
Comment (ktlim) [reply recipe/meta.yaml]: There wasn't an RFC/ticket for it, but sure.

File: recipe/meta.yaml (hunk 2 above)
Comment (erykoff) [reply recipe/meta.yaml:151]: This was required because of the psfex problem. It should be optional and conda forge controllable I think

**Comment by ktlim** [comment]
Compiler versions in conda-forge-pinning-feedstock are 12 (Linux) and 16 (OSX).  FORTRAN is 12 on both.  llvm_openmp goes to 16 (OSX only).  Boost is still 1.78.0.  cfitsio goes to 4.3.0.  log4cxx is still 0.11.0; should we keep that or move up?

**Comment by timj** [comment]
I think we wanted to update log4cxx this time and I think we need to fix `log` at the same time because of an API change (lsst/log#67)

**Pull request merged**